test/test-helpers/did-render.ts

```diff
@@ -0,0 +1,13 @@
+import Application from '../../src/application';
+
+async function didRender(app: Application): Promise<void> {
+  return new Promise<void>(resolve => {
+    let watcher = setInterval(function() {
+      if (app['_scheduled']) return;
```

Comment:
Also, yes, I'm happy that this is just an API to be used by test helpers we provide. That gives us the flexibility to change as needed in the future. I'm just mentioning these things because I think they will _eventually_ be an issue for us, and I want us to at least have a plan.